src/overlays/actors/ovl_En_Door_Etc/z_en_door_etc.c

```diff
@@ -27,42 +32,223 @@ const ActorInit En_Door_Etc_InitVars = {
+    EnDoorEtc* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+/**
+ * Calculates if the distance between `a` and `b` are greater than distance `c`
+ */
+s32 EnDoorEtc_IsDistanceGreater(Vec3f* a, Vec3f* b, f32 c) {
+    f32 dx = b->x - a->x;
+    f32 dy = b->y - a->y;
+    f32 dz = b->z - a->z;
+
+    return ((SQ(dx) + SQ(dy) + SQ(dz)) < (SQ(c)));
+}
+
+void func_80AC20A8(EnDoorEtc* this, PlayState* play) {
+    if (Object_IsLoaded(&play->objectCtx, this->objectIndex)) {
+        this->actor.flags &= ~FLAGS;
+        this->actor.objBankIndex = this->objectIndex;
+        this->actionFunc = func_80AC2354;
+        this->actor.draw = EnDoorEtc_Draw;
+    }
+}
+
+void func_80AC2118(EnDoorEtc* this, PlayState* play) {
+    if (this->angle < 0) {
+        this->angle += 0x100;
+    } else {
+        this->angle = 0;
+        this->actionFunc = func_80AC21A0;
+        this->unk_1F4 &= 0xFFFE;
+    }
+}
+
+void func_80AC2154(EnDoorEtc* this, PlayState* play) {
+    if (this->timer > 0) {
+        this->timer--;
+    } else if (this->angle >= -0x3FFF) {
+        this->angle -= 0x800;
+    } else {
+        this->angle = -0x4000;
+        this->actionFunc = func_80AC2118;
+    }
+}
+
+void func_80AC21A0(EnDoorEtc* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    Vec3f playerOffsetFromDoor;
+    s32 pad;
+    s16 var_v0;
+    s32 var_v1;
+
+    Actor_OffsetOfPointInActorCoords(&this->actor, &playerOffsetFromDoor, &player->actor.world.pos);
+    if (this->unk_1A1 == 0) {
+        if ((Player_InCsMode(play) == 0) &&
+            ((fabsf(playerOffsetFromDoor.y) < 20.0f) && fabsf(playerOffsetFromDoor.x) < 20.0f) &&
+            (fabsf(playerOffsetFromDoor.z) < 50.0f)) {
+            var_v0 = player->actor.shape.rot.y - this->actor.shape.rot.y;
+            if (playerOffsetFromDoor.z > 0.0f) {
+                var_v0 = 0x8000 - var_v0;
+            }
+            if (var_v0 < 0) {
+                var_v1 = -var_v0;
+            } else {
+                var_v1 = var_v0;
+            }
+            if (var_v1 < 0x3000) {
+                player->doorDirection = playerOffsetFromDoor.z >= 0.0f ? 1.0f : -1.0f;
+                player->doorActor = &this->actor;
+                player->doorType = -1;
+            }
+        }
+    }
+    if ((this->actor.textId == 0x239B) && (Flags_GetSwitch(play, ENDOORETC_GET_SWITCHFLAG(&this->actor)) != 0)) {
+        Flags_UnsetSwitch(play, ENDOORETC_GET_SWITCHFLAG(&this->actor));
+        this->actionFunc = func_80AC2154;
+        this->actor.textId = 0x1800; // "It won't budge!"
+        this->unk_1F4 |= 1;
+        this->timer = 0x5A;
+    }
+}
+
+void func_80AC2354(EnDoorEtc* this, PlayState* play) {
+    Actor* var_s0 = play->actorCtx.actorLists[0xA].first;
+
+    while (var_s0 != NULL) {
+        if ((var_s0->id != 5) ||
+            (EnDoorEtc_IsDistanceGreater(&var_s0->world.pos, &this->actor.world.pos, 10.0f) == 0)) {
```

Comment:
This function returns a boolean so you don't need to compare to 0
```suggestion
            !EnDoorEtc_IsDistanceGreater(&var_s0->world.pos, &this->actor.world.pos, 10.0f)) {
```